tensorflow_addons/image/filters_test.py

```diff
@@ -109,3 +110,89 @@ def test_three_channels(self):
 
 if __name__ == "__main__":
     tf.test.main()
+
+    
+@test_utils.run_all_in_graph_and_eager_modes
+class MeanFilter2dTest(tf.test.TestCase):
+    def _validateMean_2d(self, inputs, expected_values, filter_shape=(3, 3)):
+
+        values_op = mean_filter2d(inputs)
+        with self.test_session(use_gpu=False) as sess:
+            if tf.executing_eagerly():
+                expected_values = expected_values.numpy()
+                values = values_op.numpy()
+            else:
+                expected_values = expected_values.eval()
+                values = values_op.eval()
+            self.assertShapeEqual(values, inputs)
+            self.assertShapeEqual(expected_values, values_op)
+            self.assertAllClose(expected_values, values)
+
+    def testfiltertuple(self):
+        tf_img = tf.zeros([3, 4, 3], tf.int32)
+
+        with self.assertRaisesRegexp(TypeError,
+                                     'Filter shape must be a tuple'):
+            mean_filter2d(tf_img, 3)
+            mean_filter2d(tf_img, 3.5)
+            mean_filter2d(tf_img, 'dt')
+            mean_filter2d(tf_img, None)
+
+        filter_shape = (3, 3, 3)
+        msg = 'Filter shape must be a tuple of 2 integers. ' \
+              'Got %s values in tuple' % len(filter_shape)
+        with self.assertRaisesRegexp(ValueError, msg):
+            mean_filter2d(tf_img, filter_shape)
+
+        with self.assertRaisesRegexp(TypeError,
+                                     'Size of the filter must be Integers'):
+            mean_filter2d(tf_img, (3.5, 3))
+            mean_filter2d(tf_img, (None, 3))
+
+    def testfiltervalue(self):
+        tf_img = tf.zeros([3, 4, 3], tf.int32)
+
+        with self.assertRaises(ValueError):
+            mean_filter2d(tf_img, (4, 3))
+
+    def testDimension(self):
+        tf.compat.v1.disable_eager_execution()
```

Comment:
Why does this need to be disabled? and Could you re-write this test so it tests against eager and graph? The if-else method you've used in other tests is acceptable. Could you please do that for the median filter as well.

Reply:
The eager mode was only deactivated for the particular test of checking dimension. We had to deactivate eager mode for this test because placeholder is not supported in the Eager mode of execution. I needed placeholder to create tensor with None Dimension. I found no other way to create None dimensional tensor. Hence, for this particular test eager mode is deactivated.